code/code/misc/wiz_data.cc

```diff
@@ -103,72 +64,39 @@ void TMonster::wizFileSave()
 
 void TPerson::wizFileSave()
 {
-  FILE *fp;
-  sstring buf;
   Descriptor *d = NULL;
-  wizSaveData saveData;
 
   if (!(d = desc))
     return;
   if (d->connected != CON_PLYNG)  // semi arbitrary, but here for sanity
     return;
 
-  buf = format("immortals/%s/wizdata") % getName();
-  unlink(buf.c_str());
-  if (!(fp = fopen(buf.c_str(), "wa+"))) {
-    buf = format("immortals/%s") % getName();
-    if (!(fp = fopen(buf.c_str(), "r"))) {	// no immort directory 
-      if (mkdir(buf.c_str(), 0770))
-	sendTo("Unable to create a wizard directory for you.  Tell Brutius or Batopr.\n\r");
-      else
-	sendTo("Wizard directory created... type 'save' again.\n\r");
-    } else
-      fclose(fp);	// player has no wizard data, but has immort directory 
-
-    return;
-  }
+  TDatabase db(DB_SNEEZY);
 
-  saveData.setsev      = d->severity;
-  saveData.office      = d->office;
-  saveData.blockastart = d->blockastart;
-  saveData.blockaend   = d->blockaend;
-  saveData.blockbstart = d->blockbstart;
-  saveData.blockbend   = d->blockbend;
+  db.query("replace into wizdata (setsev, office, blockastart, blockaend, blockbstart, blockbend, player_id) "
+            "values (%i, %i, %i,%i, %i, %i, %i)", d->severity, d->office, d->blockastart, 
+            d->blockaend, d->blockbstart, d->blockbend, getPlayerID());
 
-  fwrite(&saveData, sizeof(wizSaveData), 1, fp);   
-  fclose(fp);
 }
 
 
 void TBeing::doOffice(sstring arg)
 {
-  FILE *fp;
-  sstring buf, buf2, buf3;
-  Descriptor *d = NULL;
-  wizSaveData saveData;
-
+  sstring buf;
   // don't use isImmortal, save always
-  if (!(GetMaxLevel() > MAX_MORT) || !(d = desc) || !hasWizPower(POWER_GOTO))
+  if (!(GetMaxLevel() > MAX_MORT) || !hasWizPower(POWER_GOTO))
     return;
 
   if (arg == "") {
     sendTo("You must supply a name as an argument (case-sensitive).\n\r");
     return;
   }
 
-  buf = format("immortals/%s/wizdata") % arg;
-  fp = fopen(buf.c_str(), "r");
-  if (!fp) {
-	  sendTo(format("Unable to open file for %s (case sensitive!).\n\r") % arg);
-    return;
-  }
-  if (fread(&saveData, sizeof(saveData), 1, fp) != 1) {
-    vlogf(LOG_BUG, format("Corrupt wiz save file for %s") % arg);
-    fclose(fp);
-    return;
-  } 
-  fclose(fp);
-  buf3 = format("The office of %s is %d.\n\r") % arg % saveData.office;
-  sendTo(buf3);
+  TDatabase db(DB_SNEEZY);
+
+  db.query("select office from wizdata where player_id = (select id from player where name = '%s')", arg.c_str());
```

Comment:
My twitch reaction for this task would be
```sql
select office from wizdata w join player_id p on w.player_id = p.player_id where p.name = '%s'
```
Can't justify it, though, your way is totally fine.